packages/ses/src/error/note-log-args.js

```diff
@@ -118,49 +123,78 @@ export const makeLRUCacheMap = budget => {
     return cell;
   };
 
+  /**
+   * @param {K} key
+   */
   const has = key => touchCell(key) !== undefined;
   freeze(has);
 
+  /**
+   * @param {K} key
+   */
   // TODO Change to the following line, once our tools don't choke on `?.`.
   // See https://github.com/endojs/endo/issues/1514
-  // const get = key => touchCell(key)?.value;
+  // const get = key => touchCell(key)?.data?.get(key);
   const get = key => {
     const cell = touchCell(key);
-    return cell && cell.value;
+    return cell && cell.data && cell.data.get(key);
   };
   freeze(get);
 
+  /**
+   * @param {K} key
+   * @param {V} value
+   */
   const set = (key, value) => {
-    if (budget >= 1) {
-      let cell = touchCell(key);
-      if (cell !== undefined) {
-        cell.value = value;
-      } else {
-        if (size >= budget) {
-          const condemned = head.prev;
-          spliceOut(condemned); // Drop least recently used
-          map.delete(condemned.key);
-          size -= 1;
-        }
-        size += 1;
-        cell = makeSelfCell(key, value);
-        map.set(key, cell);
-        spliceAfter(head, cell); // start most recently used
+    if (budget < 1) {
+      // eslint-disable-next-line no-use-before-define
+      return lruCacheMap; // Implements WeakMap.set
+    }
+
+    let cell = touchCell(key);
+    if (cell === undefined) {
+      cell = makeSelfCell(undefined);
+      spliceAfter(head, cell); // start most recently used
+    }
+    if (!cell.data) {
+      // Either a new cell or a condemned cell.  Add its data.
+      cell.data = new WeakMap();
+      // Advertise the cell for this key.
+      keyToCell.set(key, cell);
+      while (size >= budget) {
+        const condemned = head.prev;
+        spliceOut(condemned); // Drop least recently used
+        condemned.data = undefined;
+        size -= 1;
       }
+      size += 1;
```

Comment:
This is totally on me, but it would make more sense to move `size += 1` at the top of this `if` block since we've already inserted the cell, and change the `size >= budget` check to `size > budget`.